src/NServiceBus.Core/Transports/Msmq/MsmqFailureInfoStorage.cs

```diff
@@ -67,6 +67,11 @@ public bool TryGetFailureInfoForMessage(string messageId, out ProcessingFailureI
 
         public void ClearFailureInfoForMessage(string messageId)
         {
+            if (string.IsNullOrEmpty(messageId))
```

Comment:
It looks like the check should be out where the new `ClearFailureInfoForMessage` was added. It should be a check if the `message` is null there, not if the `message.Id` null. Will fix.